alma/lib/Utils/CartDataHelper.php

```diff
@@ -0,0 +1,368 @@
+                    'cart_totals' => $this->getPurchaseAmount(),
+                    'cart_totals_high_precision' => number_format($this->getPurchaseAmount(), 16),
+                ],
+                'locale' => $this->getLocale(),
+            ],
+            'customer' => $this->getCustomerData(),
+        ];
+    }
+
+    /**
+     * Get array data customer
+     *
+     * @return array getCustomerData
+     */
+    private function getCustomerData()
+    {
+        return [
+            'first_name' => $this->customer->firstname,
+            'last_name' => $this->customer->lastname,
+            'email' => $this->customer->email,
+            'birth_date' => $this->getBirthday(),
+            'addresses' => $this->getAddressesData(),
+            'phone' => $this->getPhone(),
+            'country' => $this->getCountryAddressByType('billing'),
+            'county_sublocality' => null,
+            'state_province' => $this->getStateProvince('billing'),
+        ];
+    }
+
+    /**
+     * Get birthday date
+     *
+     * @return string|null
+     */
+    private function getBirthday()
+    {
+        $birthday = $this->customer->birthday;
+        if ($birthday == '0000-00-00') {
+            return null;
+        }
+
+        return $birthday;
+    }
+
+    /**
+     * Get addresses of customer Prestashop
+     *
+     * @return array getCustomerAddresses
+     */
+    private function getCustomerAddresses()
+    {
+        if (version_compare(_PS_VERSION_, '1.5.4.0', '<')) {
+            $idLang = $this->context->language->id;
+        } else {
+            $idLang = $this->customer->id_lang;
+        }
+
+        return $this->customer->getAddresses($idLang);
+    }
+
+    /**
+     * Get addresses of customer for Alma
+     *
+     * @return array getAddressesData
+     */
+    private function getAddressesData()
+    {
+        $addresses = [];
+        
+        foreach ($this->getCustomerAddresses() as $address) {
+            array_push($addresses, [
+                'line1' => $address['address1'],
+                'postal_code' => $address['postcode'],
+                'city' => $address['city'],
+                'country' => Country::getIsoById((int) $address['id_country']),
+                'county_sublocality' => null,
+                'state_province' => $address['state'],
+            ]);
+        }
+
+        return $addresses;
+    }
+
+    /**
+     * get phone between info customer and addresses customer
+     *
+     * @return string
+     */
+    private function getPhone()
+    {
+        $phone = null;
+        $shippingAddress = $this->getCartAddressByType('shipping');
+
+        if ($shippingAddress->phone) {
+            $phone = $shippingAddress->phone;
+        } elseif ($shippingAddress->phone_mobile) {
+            $phone = $shippingAddress->phone_mobile;
+        }
+
+        if (is_null($phone)) {
```

Comment:
if there is no phone, a phone will be requested later in the checkout.
this is a little bit intrusive to retrieve a phone number from an old address

Reply:
It was like that before, you think we will change it ?